effectful/handlers/llm/__init__.py

```diff
@@ -1,17 +1,33 @@
 import dataclasses
 import functools
 import inspect
-from collections.abc import Callable, Iterable
+import types
+import weakref
+from collections import ChainMap
+from collections.abc import Callable, Mapping
+from typing import Any
 
 from effectful.ops.syntax import defop
 from effectful.ops.types import NotHandled, Operation
 
 
 @dataclasses.dataclass(frozen=True)
 class Template[**P, T]:
-    __signature__: inspect.Signature
     __prompt_template__: str
-    tools: tuple[Operation, ...]
+    __signature__: inspect.Signature
+    __name__: str = ""
+    lexical_context: Mapping[str, Any] = dataclasses.field(
```

Comment:
A more specific annotation `lexical_context: weakref.WeakValueDictionary[str, Any]` should work, right?